### PR TITLE
fix: add an exception if executable not found

### PR DESCRIPTION
**Pull request by dipinknair**
Currently, if no executable found in default location when using `get_mechanical_path` , then we are returning `None`. This is causing issue when we call `version_from_path` where we give `path` as input. We do not have any check if path is None. 

Alternately we can avoid this exception and return `path` as empty string (as it is) it is instead of `None`. If that is the case then  `version_from_path` will take care of value error if `path` is `empty string`

### CODE REVIEW

File: src/ansys/tools/path/path.py [hunk 1]
```diff
@@ -967,6 +967,8 @@ def _get_application_path(
             if (exe_loc, exe_version) != ("", ""):  # executable not found
                 if os.path.isfile(exe_loc):
                     return exe_loc
+            else:
+                raise Exception("{product} executable not found in default locations")
```
Comment (RobPasMue) [inline src/ansys/tools/path/path.py:971]: Why would you prefer to raise an Exception? I would rather handle properly the ``None`` value.

File: src/ansys/tools/path/path.py [hunk 2]
```diff
@@ -967,6 +967,8 @@ def _get_application_path(
             if (exe_loc, exe_version) != ("", ""):  # executable not found
                 if os.path.isfile(exe_loc):
                     return exe_loc
```
Comment (koubaa) [inline src/ansys/tools/path/path.py:969]: as discussed this can be moved somewhere else

**Review by koubaa — CHANGES_REQUESTED** [review]
The message of the exception isn't actionable by the end user

**Review by RobPasMue — APPROVED** [review]
LGTM - although we could actively check for None as well. But this is more than enough

**Comment by RobPasMue** [comment]
We should do a new release after #193 - thanks @nmalsang - what's the status here? We could include this PR as well.

File: src/ansys/tools/path/path.py [hunk 3]
```diff
@@ -1132,6 +1132,8 @@ def version_from_path(product: PRODUCT_TYPE, path: str) -> int:
         Integer version number (for example, 231).
 
     """
+    if not isinstance(path, str):
+        raise ValueError("Provided path is not a string")
```
Comment (germa89) [inline src/ansys/tools/path/path.py:1136]:
```suggestion
        raise ValueError(f"Provided path '{path}' is not a string.")
```

**Review by germa89 — CHANGES_REQUESTED** [review]
LGTM. But I think the message could be more explicative.

**Pull request merged**